### PR TITLE
DOC: Several docstrings

### PR DESCRIPTION
**Pull request by FranzForstmayr**

### CODE REVIEW

**Comment by FranzForstmayr** [comment]
related to #7168

File: scipy/signal/filter_design.py [hunk 1]
```diff
@@ -1785,15 +1871,54 @@ def lp2bs(b, a, wo=1.0, bw=1.0):
...
+
+    >>> fs = 100
+    >>> bf = 2 * np.pi * np.array([7, 13])
+    >>> filts = signal.lti(*signal.butter(4, bf, btype='bandpass', analog=True))
+    >>> filtz = signal.lti(*signal.bilinear(filts.num, filts.den, fs))
+    >>> wz, hz = signal.freqz(filtz.num, filtz.den)
+    >>> ws, hs = signal.freqs(filts.num, filts.den, worN=fs*wz)
+
+    >>> plt.semilogx(wz*fs/(2*np.pi), np.abs(hz), label=r'$|H(j \omega)|$ of analog bandpass')
+    >>> plt.semilogx(wz*fs/(2*np.pi), np.abs(hs), label=r'$|H_z(j \omega)|$ of digital bandpass')
```
Comment (ilayn) [inline scipy/signal/filter_design.py]: This legend entry should read `'$|H_z(e^{j \omega})|$ of digital bandpass'` for the discrete time system. But I think you can simply drop the analog/digital bandpass as the context is clear. 

Same applies to bilinear_zpk below too.

File: scipy/signal/ltisys.py [hunk 2]
```diff
@@ -3414,6 +3414,19 @@ def dimpulse(system, x0=None, t=None, n=None):
...
 
+    Examples
+    --------
+    >>> from scipy import signal
+    >>> import matplotlib.pyplot as plt
+    >>> import numpy as np
+
+    >>> butter = signal.dlti(*signal.butter(3, 0.5))
+    >>> t, y = signal.dimpulse(butter, n=25)
+    >>> plt.stem(t, np.squeeze(y), basefmt=" ")
```
Comment (ilayn) [inline scipy/signal/ltisys.py]: For these plots matplotlib has [`plt.step` function](https://matplotlib.org/gallery/lines_bars_and_markers/step_demo.html#sphx-glr-gallery-lines-bars-and-markers-step-demo-py) which look better in my opinion

File: scipy/signal/ltisys.py [hunk 3]
```diff
@@ -3488,6 +3501,18 @@ def dstep(system, x0=None, t=None, n=None):
...
 
+    Examples
+    --------
+    >>> from scipy import signal
+    >>> import matplotlib.pyplot as plt
+    >>> import numpy as np
+
+    >>> butter = signal.dlti(*signal.butter(3, 0.5))
+    >>> t, y = signal.dstep(butter, n=25)
+    >>> plt.stem(t, np.squeeze(y), basefmt=" ")
```
Comment (ilayn) [inline scipy/signal/ltisys.py]: Same as dimpulse comment.

File: scipy/signal/filter_design.py [hunk 4]
```diff
@@ -2216,6 +2341,24 @@ def bilinear_zpk(z, p, k, fs):
...
+    >>> import numpy as np
+
+    >>> fs = 100
+    >>> bf = 2 * np.pi * np.array([7, 13])
+    >>> filts = signal.lti(*signal.butter(4, bf, btype='bandpass', analog=True, output='zpk'))
+    >>> filtz = signal.lti(*signal.bilinear_zpk(filts.zeros, filts.poles, filts.gain, fs))
+    >>> wz, hz = signal.freqz_zpk(filtz.zeros, filtz.poles, filtz.gain)
+    >>> ws, hs = signal.freqs_zpk(filts.zeros, filts.poles, filts.gain, worN=fs*wz)
+    >>> plt.semilogx(wz*fs/(2*np.pi), np.abs(hz), label=r'$|H(j \omega)|$ of analog bandpass')
+    >>> plt.semilogx(wz*fs/(2*np.pi), np.abs(hs), label=r'$|H_z(j \omega)|$ of digital bandpass')
```
Comment (ilayn) [inline scipy/signal/filter_design.py]: See the comment for `bilinear`

**Review by ilayn — COMMENTED** [review]
Thanks. This looks good to me except a few minor details.

File: scipy/signal/filter_design.py [hunk 5]
```diff
@@ -1785,15 +1871,54 @@ def lp2bs(b, a, wo=1.0, bw=1.0):
 
 
 def bilinear(b, a, fs=1.0):
-    """Return a digital filter from an analog one using a bilinear transform.
+    """
```
Comment (ilayn) [inline scipy/signal/filter_design.py]: This one also requires an `r` identifier for the `\omega`

**Comment by FranzForstmayr** [comment]
I'll fix this as soon as possible!

**Comment by ilayn** [comment]
Looks nice. 

https://11286-1460385-gh.circle-artifacts.com/0/html-scipyorg/generated/scipy.signal.bilinear.html#scipy.signal.bilinear

The only remaining detail is that the discrete time filter responses are evaluated on the unit circle instead of the imaginary axis Hence it needs to be `H_z(e^{j \omega})` .

**Comment by FranzForstmayr** [comment]
@ilayn ,
 can you help me, what's wrong in this case?

**Comment by rgommers** [comment]
nothing is wrong, the Appveyor failure is an unrelated flake

File: scipy/signal/filter_design.py [hunk 6]
```diff
@@ -1785,15 +1871,54 @@ def lp2bs(b, a, wo=1.0, bw=1.0):
...
     See Also
     --------
     lp2lp, lp2hp, lp2bp, lp2bs
     bilinear_zpk
 
+    Examples
+    --------
+    >>> from scipy import signal
+    >>> import matplotlib.pyplot as plt
+    >>> import numpy as np
```
Comment (rgommers) [inline scipy/signal/filter_design.py]: nitpick: `import numpy` is never needed in examples, it's assumed

**Comment by ilayn** [comment]
Looks good, thank you @FranzForstmayr !

**Pull request merged**